### DISCUSSION TITLE
How best to install R packages not on conda?

### DISCUSSION THREAD

**Discussion by Robinlovelace** [prefix-dev/pixi#2082] [Q&A] [answered]
Let's say we want to install the geocompkg package? With the `pak` pkg that would be

```r
pak::pkg_install("geocompx/geocompkg")
```

Have tried this but no luck so far:

```
pixi shell R -e 'pak::pkg_install("geocompx/geocompkg")'
```

**Comment by wolfv** [comment]
@Robinlovelace can you elaborate what kind of errors you are seeing?

**Reply by Robinlovelace** [reply]
Working on it... context: https://github.com/geocompx/docker/pull/54

As you'll see there we're testing pixi's Docker containers for a cross-language setup.

**Reply by Robinlovelace** [reply]
Good news: issue is fixed! 

https://github.com/geocompx/docker/blob/984cdded697768d7a2eb1d215f950611f242afaa/pixi-r/Dockerfile#L11

FYI that pixi-based image is now available. Would be grateful if anyone could test with:

```
docker run -it ghcr.io/geocompx/docker:r-pixi bash
```

**Comment by Robinlovelace** [comment, marked as answer]
Short answer:

```
RUN pixi run R -e 'pak::pkg_install("geocompx/geocompkg")'
```

**Comment by Robinlovelace** [comment]
I have another question, will open another question..

**Comment by pakiessling** [comment]
@Robinlovelace @wolfv 
Do you have a recommendation for reproducing enviroments with pak / devtools packages in them (e.g sharing a pixi.toml with other people) Is there a better way than including the pak installation steps as pixi tasks?

**Reply by jjjermiah** [reply]
you could create an activation script that will get run on any environment activation

**Comment by pakiessling** [comment]
Ah that's a great idea, thank you!
________________________________

**Comment by Robinlovelace** [comment]
Hi all, just picking up on this script, can anyone advise on how to integrate the 

```
RUN pixi run R -e 'pak::pkg_install("geocompx/geocompkg")'
```

call in the pixi.toml file? I'm looking at https://prefix.dev/docs/pixi/configuration#the-activation-table

**Reply by Robinlovelace** [reply]
That fails.

**Comment by Robinlovelace** [comment]
Trying the following:

```
[activation]
scripts = ["env_setup.sh"]
```

With env_setup.sh containing:

```sh
Rscript -e "pak::pak('itsleeds/tds', ask = FALSE)"
```

**Reply by Robinlovelace** [reply]
That approach failed to install the dependencies.

**Reply by ruben-arts** [reply]
Is there an indication why this fails? Not that on windows you need to use `.bat` files.

**Reply by Robinlovelace** [reply]
Not sure why it fails. I'm switching to Docker to ensure reproducibility and ease installation.
